src/project/service.ts

```diff
@@ -79,27 +80,36 @@ function mapRecord(input: ProjectsRecord, client: Client): Project {
 
 }
 
-export async function  addPersonToProject(params: PersonProjectForm): Promise<Person> {
+export async function  addPersonToProject(params: PersonProjectForm, projectId: number, origin: string): Promise<void> {
 
   const principalUri = await findOrCreatePrincipal(params.href, params.name);
 
-  let person : Person;
+  let project : Project;
   try {
-    person = await personService.findByPrincipalUrl(principalUri);
+    await personService.findByPrincipalUrl(principalUri);
+    project = await findById(projectId);
   } catch(error) {
 
     if(!(error instanceof NotFound)){
       throw error;
     }
 
 
-    person = await personService.create({
+    await personService.create({
       name: params.name,
       principalUri,
     });
-    return person;
+    project = await findById(projectId);
+
+  } finally {
```

Comment:
This is not a good use of finally. Just leave this outside of this block. It now runs even if there's other errors happening, and we really don't want that.